### ISSUE TITLE
TCK testing with GlassFish 7 gets stuck during ejb30/lite/packaging/embed/provider testing

### ISSUE THREAD

**Issue by scottmarlow** [jakartaee/platform-tck#1110]
Run only the `ejb30/lite/packaging/embed/provider` tests and you will see:

```
[runcts] OUT => [javatest.batch] Beginning Test:  com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed
   [runcts] OUT => [javatest.batch] ********************************************************************************
   [runcts] OUT => [javatest.batch] 
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - harness.executeMode is set to "0"
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - AutoDeployment.init()
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - AutoDeployment:  Using porting set #1
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - Using porting.ts.deploy2.class.1
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - AutoDeployment.init()
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - AutoDeployment:  Using porting set #2
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - Using porting.ts.deploy2.class.2
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - ss:  modified runtime.xml
   [runcts] OUT => [javatest.batch] getCommonApps - relativeTestDir = com/sun/ts/tests/ejb30/lite/packaging/embed/provider
   [runcts] OUT => [javatest.batch] Loading /root/jakartaeetck/bin/xml/../../src/com/sun/ts/lib/harness/commonarchives.properties
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - Undeploying apps...
   [runcts] OUT => [javatest.batch] Undeployed apps from - /root/jakartaeetck/bin/xml/../../dist/com/sun/ts/tests/ejb30/lite/packaging/embed/provider
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - Deploying apps for forward rebuildable...
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - There are no archives to deploy in this directory:  /root/jakartaeetck/bin/xml/../../dist/com/sun/ts/tests/ejb30/lite/packaging/embed/provider
   [runcts] OUT => [javatest.batch] Deployed apps from - /root/jakartaeetck/bin/xml/../../dist/com/sun/ts/tests/ejb30/lite/packaging/embed/provider
   [runcts] OUT => [javatest.batch] Undeploying apps...
   [runcts] OUT => [javatest.batch] Deploying apps for forward rebuildable...
   [runcts] OUT => [javatest.batch] There are no archives to deploy in this directory:  /root/jakartaeetck/bin/xml/../../dist/com/sun/ts/tests/ejb30/lite/packaging/embed/provider
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - harness.executeMode is set to "0"
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:25:  Harness - keywords (to be passed to tests) set to:  all
   [runcts] OUT => [javatest.batch] 08-10-2022 18:38:26:  Harness - sClassPathFromExecProps = /root/jakartaeetck/bin/xml/../../dist/com/sun/ts/tests/ejb30/lite/packaging/embed/provider/ejbembed_vehicle_ejb.jar
   [runcts] OUT => [javatest.batch] command: com.sun.ts.lib.harness.ExecTSTestCmd CLASSPATH=/root/jakartaeetck/bin/xml/../../lib/tsharness.jar:/root/jakartaeetck/bin/xml/../../lib/cts.jar:/root/jakartaeetck/bin/xml/../../lib/glassfishporting.jar:/root/jakartaeetck/bin/xml/../../lib/commons-lang3-3.3.2.jar:/root/vi/glassfish7/glassfish/../javadb/lib/derbyclient.jar:/root/vi/glassfish7/glassfish/../javadb/lib/derbyshared.jar:/root/vi/glassfish7/glassfish/../javadb/lib/derbytools.jar:/root/vi/glassfish7/glassfish/lib/embedded/glassfish-embedded-static-shell.jar:/root/jakartaeetck/bin/xml/../../dist/com/sun/ts/tests/ejb30/lite/packaging/embed/provider/ejbembed_vehicle_ejb.jar DISPLAY=:0.0 HOME=/ TMP= windir= SYSTEMROOT= /opt/jdk-11.0.2/bin/java -Dcts.tmp=/root/jakartaeetck/bin/xml/../../tmp -Djava.util.logging.config.file=/root/vi/glassfish7/glassfish/domains/domain1/config/logging.properties -Dtest.ejb.stateful.timeout.wait.seconds=480 -Ddeliverable.class=com.sun.ts.lib.deliverable.cts.CTSDeliverable com.sun.ts.tests.common.vehicle.VehicleClient -p /root/jakartaeetck/bin/xml/../../tmp/tstest.jte -t customProvider -vehicle ejbembed
   [runcts] OUT => [javatest.batch] ************************************************************
   [runcts] OUT => [javatest.batch] * props file set to "/tmp/1003270000-cts-props.txt"
   [runcts] OUT => [javatest.batch] ************************************************************
   [runcts] OUT => [javatest.batch] Timeout signaled after 1,200 seconds
   [runcts] OUT => [javatest.batch] Test not responding after 100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 900 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,000 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 1,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,900 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 2,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,900 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,000 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 3,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,900 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,000 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 4,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,900 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,000 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 5,000 interrupts
```

**Comment by scottmarlow** [comment]
I'm trying to recreate locally via https://github.com/eclipse-ee4j/jakartaee-tck/wiki/Instructions-for-running-the-nightly-engineering-build-of-Platform-TCK-with-Podman-(currently-for-EE-10)

**Comment by scottmarlow** [comment]
I can recreate it locally.  I don't yet have access to GlassFish server logs but hopefully that will change in a few hours when the tests complete.

Some output that I am seeing:
```
   [runcts] OUT => [javatest.batch] Test not responding after 1,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 1,900 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,000 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 2,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 2,900 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,000 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 3,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 3,900 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 4,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 4,900 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 5,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,000 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,100 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,200 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,300 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,400 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,500 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,600 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,700 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,800 interrupts
   [runcts] OUT => [javatest.batch] Test not responding after 5,900 interrupts
   [runcts] OUT => [javatest.batch] Test com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed has timed out and is not responding after 6,000 interrupts
```

**Comment by scottmarlow** [comment]
Click  [here to via the Client_customProvider_from_ejbembed.jtr](https://github.com/eclipse-ee4j/jakartaee-tck/files/9305393/Client_customProvider_from_ejbembed.jtr.txt)

**Comment by gurunrao** [comment]
@scottmarlow -August 3/4 runs with JDK11/17 doesn't have timeouts as per logs - https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-run-100/122/, https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-run-100/123/
@arjantijms  - was there any change with GF last week, which will cause  interrupts with `com/sun/ts/tests/ejb30/lite/packaging/embed/provider/Client.java#customProvider_from_ejbembed` tests?

**Comment by dmatej** [comment]
First failing ejb30 test on my computer failed because of helidon. Is this line in your logs too?
`Caused by: java.lang.UnsupportedClassVersionError: io/helidon/config/ConfigException has been compiled by a more recent version of the Java Runtime (class file version 61.0), this version of the Java Runtime only recognizes class file versions up to 55.0`

Later failed more tests, I just executed whole ejb30 at night and interrupted them after this:

`[runcts] OUT => [javatest.batch] Test not responding after 118,300 interrupts`
```
Number of tests completed:  2420 (2240 passed, 180 failed, 0 with errors)
[runcts] OUT => [javatest.batch] Number of tests remaining:  1503
```
I will try to fix this today, probably replace that jdk17 profile with some classloader solution (exclude helidon if we are running on java before JDK17).

Damn, I should also push my tck runner, there's no complicated setup required to run "old" TCK tests ...

**Comment by scottmarlow** [comment]
I tried `grep -rli "UnsupportedClassVersionError"` and got no matches from the folder I am doing local testing from.  

Also no matches from `grep -rli io.helidon.config.ConfigException` as well.  @dmatej which log file do you see the UnsupportedClassVersionError failure in?  

I started a JDK 17 (ejb30/lite/packaging/embed/provider) test run on https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-run-100/129.

**Comment by scottmarlow** [comment]
@dmatej The [Jenkins test run with JDK17](https://ci.eclipse.org/jakartaee-tck/job/10/job/eftl-jakartaeetck-run-100/129/) is still timing out the same it seems (hasn't finished yet but definitely not running correctly).

Which GlassFish 7 jar has `helidon.config.ConfigException` in it?  

I tried `grep -rli helidon.config.ConfigException` to find the suspect class in my local GlassFish 7 (nightly build) but got no matches.  Do you know which GlassFish file contains the `helidon.config.ConfigException`?  I would like to check the Java version number for that file on my machine.

**Comment by dmatej** [comment]
Slow down a bit ... or not :D
If you build GF with JDK17, it enables a profile to add helidon. Helidon has JDK17 as a minimum. Then GF could not pass tests on JDK11 - I fixed that locally now, instead of the profile (which leads to two variants of the server, with or without microprofile support) now it uses a classloader filter; when GF runs on JDK11, the extension is not loaded (before my fix it printed errors on logs; also - I noticed that an the server was just a `Could not load service class ` message, on the TCK side were stacktraces).

That resolves my issue with timeouts, but as I read your comments, there is probably yet another issue. I will take a look now. If it results in the same situation, it is probably something related, even if with different cause, probably in same classes.

**Comment by dmatej** [comment]
I have it reproduced, just fyi. Seems the problem is not the server, but the embedded ... I will find it.

**Comment by dmatej** [comment]
... and I have a fix. This will go to the TCK, because it parasites on logging.properties of the GlassFish domain and uses them for embedded+client too, but they runs in parallel to the GlassFish domain.

Before implementing GJULE last month it did not collide, because GlassFish logging was partially configured via HK2 services which did not start on the client, but now it is configured in a standard way. That results in a state where two standalone JVMs use the same logging.properties file, ... and the same server.log file -> to my surprise the embedded cannot write to the file in parallel to the GF domain and waits instead ... indefinitely. I thought this is allowed on Linux ...?

Nevermind, I will push changes in the morning, at night I will rebuild+run it yet for the last time ...
